src/data/dataset.py

```diff
@@ -17,15 +18,21 @@ class ConvAI2Dataset(Dataset):
     _YOUR_PERSONA_PREFIX = "your persona: "
     _PARTNER_PERSONA_PREFIX = "partner's persona: "
 
-    def __init__(self, path, tokenizer, max_seq_len):
+    def __init__(self, path, tokenizer_name, max_context_len, max_target_len=None, have_candidates=True):
         self.dataset = []
         self.num_dialogs = 0
 
-        bos = tokenizer.bos_token
-        eos = tokenizer.eos_token
+        self.context_tokenizer = AutoTokenizer.from_pretrained(tokenizer_name, truncation_side="left")
+        self.candidate_tokenizer = AutoTokenizer.from_pretrained(tokenizer_name)
```

Comment:
Probably, you can specify max length here. And let's set truncation side for candidate tokenizer explicitly.